### DISCUSSION TITLE
NextJS Development Link Not Working

### DISCUSSION THREAD

**Discussion by sr5434** [community/community#119676] [Codespaces]
### Select Topic Area

Bug

### Body

When I run ```npm run dev``` in my NextJS project on Codespaces(browser), it has a link where I can see a live preview of my app. However, when I click the link, nothing happens. This behavior didn't used to exist before.

**Comment by ssanjua** [comment]
by link you mean a localhost url?
works if you type this 'link' in the browser?

**Reply by sr5434** [reply]
Yes, I meant the localhost URL. It won't work as I am using the browser and not Visual Studio code. Before, when I clicked the link, Codespaces would forward it to a special github.dev URL.

**Reply by ssanjua** [reply]
can you provide more details about your Codespaces setup or any error messages you see in the browser console when trying to access the link?

Here are some steps and considerations to potentially resolve this issue:

Check Codespaces Configuration: Ensure that the Codespaces configuration is properly set up to forward the development server port. By default, NextJS runs on port 3000, so this port needs to be exposed. You can check the port visibility in the Codespaces environment to ensure it's set to 'public'.
Manual Port Forwarding: If automatic port forwarding isn't working, you might need to set up port forwarding manually. This can usually be done in the Codespaces settings or via the Codespaces terminal by specifying which port should be exposed.
Use Direct URL: Instead of relying on the link auto-generated by npm run dev, manually construct the URL by using the forwarded port URL. This can typically be accessed via the Codespaces 'Ports' tab, where you can see the forwarded URL which is accessible from the browser.
Browser Compatibility: Check if the issue persists across different browsers. Sometimes, specific browser settings or extensions can block the functionality of such forwarded URLs.
Review Recent Changes: If this functionality worked before and recently stopped working, reviewing any changes made to the Codespaces configuration or the NextJS project settings might provide some clues. It’s possible that updates or modifications could have inadvertently affected how port forwarding is handled.
Consult Documentation or Support: If the problem persists, checking the GitHub Codespaces documentation for any updates on port forwarding and URL handling or reaching out to GitHub support might be necessary. There could be updates or changes in the platform that might affect how these URLs are managed.

**Reply by sr5434** [reply]
When I try to forward a port or click the link, it says "activating extensions" and does nothing.
